test/racingcar/car_test.go

```diff
@@ -0,0 +1,32 @@
+package racingcar_test
+
+import (
+	"testing"
+
+	"github.com/poi1649/go-racing-car/racingcar"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCarNameValidation(t *testing.T) {
+	testCases := []struct {
+		name string
+	}{
+		{""},
+		{"123456"},
+		{"abcdef"},
+		{"한글다섯글자"},
+	}
+	for _, tc := range testCases {
+		_, err := racingcar.NewCar(tc.name)
+		assert.Error(t, err, "name: %s", tc.name)
+	}
```

Comment:
P1: 실패해야 하는 테스트가 성공해버린 것 같아요. 

아래와 같은 의도셨던 것 같은데 맞나요? 아래처럼 작성하면 테스트가 실패할테니 Error와 Equal 함수 사용법만 참고해서 적절히 수정하면 좋을 것 같아요.

```suggestion
		_, err := racingcar.NewCar(tc.name)
		if assert.Error(t, err) {
			assert.Equal(t, err.Error(), fmt.Sprintf("name: %s", tc.name))
		}
```

Reply:
앗 이부분은 `assert.Error()` 통해서 에러가 발생했다만 확인을 하고 있었어서 위처럼 작성했습니다. 구현코드 주석(문서) 에서 `Error asserts that a function returned an error (i.e. not 'nil').`라고 나와있어서 그냥 에러 여부만 확인하고 끝냈었는데, 에러 내용 검증에 대한 부분을 빼먹고 안했네요... (문서의 예시 뒤에도 에러내용 검증부분이 있는데 빼먹고 안해버렸네요ㅠ)

에러 내용을 검증하면 아래처럼 해서 추천해주신것처럼 하거나, 아니면 아래처럼 `assert.EqualError`로 한번에 검증하게 될 것 같네요
```go
_, err := racingcar.NewCar(tc.name)
assert.EqualError(t, err, tc.expectedMsg, "name: %s, expected : %v, actual : %v", tc.name, tc.expectedMsg, err)
```

---

그러고 실패해야하는 테스트가 성공해버린것같다고 하셨는데, 이부분은 이해가 안되서,, 혹시 제가 놓친 케이스나 실패해야 할 부분이 어느부분인지 말씀해주실 수 있으실까요..?
자동차 이름 글자수에대해서 잘못된 수이면 예외를 잘 발생시키고 있고, 혹시나 해서 예외를 발생시키면 안되는 글자수를 테스트케이스로 추가해서 돌려봐도 정상적으로 예외가 발생 안하면서 테스트가 실패하는 모습을 잘 보여줘서. 혹시 제가 놓친부분이나, 서로 다르게 이해한 부분이 있나 해서 추가 코멘트 남깁니다!

![image](https://github.com/poi1649/go-racing-car/assets/30036534/46ff913d-6d08-410b-98dc-543c0b10f5f1)

Reply:
아아 애초부터 에러 메시지와는 별개로 에러가 nil인지만 체크하시려는 의도셨군요?
그러면 오히려 `Error` 메서드를 제대로 용도대로 사용하셨네요.

TMI: Error 메서드의 msgAndArgs 매개변수의 경우 선택사항이라 전 사실 사용하지를 않아서 조금 헷갈렸습니다.